src/renderer/views/swap/SwapView.tsx

```diff
@@ -138,49 +128,96 @@ const SuccessRouteView: React.FC<Props> = ({
     keystoreService: { keystoreState$, validatePassword$ }
   } = useWalletContext()
 
+  const { chain: sourceChain } = sourceAsset.type === AssetType.SYNTH ? mayaDetails.asset : sourceAsset
+  const { chain: targetChain } = targetAsset.type === AssetType.SYNTH ? mayaDetails.asset : targetAsset
+
+  const poolAddressThor$ = poolAddressesByChain$(sourceChain)
+  const poolAddressMaya$ = poolAddressesByChainMaya$(sourceChain)
+  const combinedPoolAddresses = useObservableState(
+    FP.pipe(
+      Rx.combineLatest([poolAddressThor$, poolAddressMaya$]).pipe(
+        RxOp.map(([thorPoolAddressRD, mayaPoolAddressRD]) =>
+          FP.pipe([thorPoolAddressRD, mayaPoolAddressRD], A.map(RD.toOption), ([thorOption, mayaOption]) => ({
+            thor: thorOption, // Include the ThorChain pool address
+            maya: mayaOption // Include the MayaChain pool address
+          }))
+        ),
+        RxOp.distinctUntilChanged(), // Avoid emitting the same values multiple times
+        RxOp.debounceTime(100) // Throttle frequent updates (adjust time as needed)
```

Comment:
It's re-renders in every 100 ms. @Thorian1te

Reply:
ok